lib/coverband/utils/coverband_results.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+####
+# A way to access the various coverage data breakdowns
+####
+module Coverband
+  module Utils
+    class CoverbandResults
+      attr_accessor :type, :results
+
+      def initialize(report)
+        self.results = {
+          Coverband::EAGER_TYPE => Coverband::Utils::Result.new(report[Coverband::EAGER_TYPE]),
+          Coverband::RUNTIME_TYPE => Coverband::Utils::Result.new(report[Coverband::RUNTIME_TYPE]),
+          Coverband::MERGED_TYPE => Coverband::Utils::Result.new(report[Coverband::MERGED_TYPE])
```

Comment:
RE the concern about memory. If `self.results` was of a different type, could have merged created only when needed? Lazily?

Reply:
yeah lazy processing could be interesting, I guess I will see how bad memory and perf is on larger projects